Calling inline proc calls proc with borrow pragma generates invalid C code

mytests/modfoo.nim:
```nim
type
  DInt* = distinct int32

proc `==`(x, y: DInt): bool {.borrow.}

func isFail*(x: DInt): bool {.inline.} =
  x == DInt 0
```

mytests/testbrw.nim:
```nim
import modfoo

discard isFail(DInt(0))
```

```console
$ ./bin/nimony c mytests/testbrw.nim 
bin/nimsem  --isSystem m nimcache/sysvq0asl.1.nif nimcache/sysvq0asl.2.nif nimcache/sysvq0asl.2.idx.nif
bin/nimsem  m nimcache/modv5te5u.1.nif nimcache/modv5te5u.2.nif nimcache/modv5te5u.2.idx.nif
bin/nimsem  --isMain m nimcache/tesxe3no4.1.nif nimcache/tesxe3no4.2.nif nimcache/tesxe3no4.2.idx.nif
cc -c   -Ivendor/mimalloc/include vendor/mimalloc/src/static.c -o nimcache/static.o
bin/hexer --bits:64 nimcache/tesxe3no4.2.nif
bin/nifc c --compileOnly --isMain  nimcache/tesxe3no4.c.nif
cc -c  -Imytests  nimcache/tesxe3no4.c -o nimcache/tesxe3no4.o
nimcache/tesxe3no4.c:407:1: error: multiple storage classes in declaration specifiers
  407 | extern static inline NB8 eqQ_0_modv5te5u(NI32 x_0, NI32 y_0);
      | ^~~~~~
nimcache/tesxe3no4.c:407:26: warning: inline function ‘eqQ_0_modv5te5u’ declared but never defined
  407 | extern static inline NB8 eqQ_0_modv5te5u(NI32 x_0, NI32 y_0);
      |                          ^~~~~~~~~~~~~~~
```

--- a/src/hexer/nifcgen.nim
+++ b/src/hexer/nifcgen.nim
@@ -660,9 +660,13 @@ proc parsePragmas(c: var EContext; n: var Cursor): CollectedPragmas =
           result.externName = pool.strings[n.litId]
           inc n
         of NodeclP, SelectanyP, ThreadvarP, GlobalP, DiscardableP, NoReturnP,
-           VarargsP, BorrowP, NoSideEffectP, NoDestroyP, ByCopyP, ByRefP,
+           VarargsP, NoSideEffectP, NoDestroyP, ByCopyP, ByRefP,
            InlineP, NoinlineP, NoInitP, InjectP, GensymP, UntypedP, ViewP,
            InheritableP, PureP:
+          result.flags.incl pk
+          inc n
+        of BorrowP:
+          result.flags.incl InlineP
           result.flags.incl pk
           inc n
         of HeaderP:
@@ -818,8 +822,6 @@ proc traverseProc(c: var EContext; n: var Cursor; mode: TraverseMode) =
   if SelectanyP in prag.flags:
     c.addKey genPragmas, "selectany", pinfo
 
-  if BorrowP in prag.flags:
-    c.addKey genPragmas, $InlineP, pinfo
   closeGenPragmas c, genPragmas
 
   skip n # miscPos

--- a/tests/nimony/sysbasics/mdistincts.nim
+++ b/tests/nimony/sysbasics/mdistincts.nim
@@ -1,0 +1,7 @@
+type
+  DInt* = distinct int32
+
+proc `==`(x, y: DInt): bool {.borrow.}
+
+func isFail*(x: DInt): bool {.inline.} =
+  x == DInt 0

--- a/tests/nimony/sysbasics/tdistincts.nim
+++ b/tests/nimony/sysbasics/tdistincts.nim
@@ -1,3 +1,5 @@
+import mdistincts
+
 type
   VarId* = distinct int
 
@@ -22,3 +24,5 @@ type
   WINBOOL* = distinct int32
 
 proc `==`(x, y: WINBOOL): bool {.borrow.}
+
+discard isFail(DInt(0))

